### ISSUE TITLE
`process_reserved_name` defined within scope of `MIR_NO_IO` but used outside

### ISSUE THREAD

**Issue by eliphatfs** [vnmakarov/mir#226]
The master does not compile when setting `MIR_NO_IO` to 1 and `MIR_NO_SCAN` to 0 now.

**Comment by vnmakarov** [comment]
OK.  I'll will look into it.

I think using MIR_NO_IO and MIR_NO_SCAN has a very limited scope and only when you need to decrease code size for some very constraint devices but the decrease is very small when MIR-generator is used.  So may be these macros are necessary at all.  I'll think what to do with them.

**Comment by eliphatfs** [comment]
Yes I've also found that with `-Os` the size difference is only 370K vs 369K (on windows) with MIR_NO_IO on and off.
